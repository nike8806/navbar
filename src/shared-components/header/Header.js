import React, { Component } from 'react';
import SearchInput from './search-input/SearchInput';
import './Header.scss';
// TODO Use the Sass in case to extend and use just the necessary class
import '@fortawesome/fontawesome-free/css/all.min.css';

/**
 * Handle the Header page
 */
class Header extends Component {
  /**
  * OnSubmit function
  */
  onIconClick = () => {
    console.log('User');
  }

  render() {
    return (
      <div className="header">
        <div className="header__menu">
          <div className="header__menu__title">
            Start Bootstrap Menu
          </div>

          <i className="fas fa-bars header__menu__bars" />
        </div>

        <div className="header__widgets">
          <SearchInput classNameComponent="header__widgets__search" />
          <button type="button" onClick={this.onIconClick}>
            <i className="fa fa-bell" />
          </button>
          <button type="button" onClick={this.onIconClick}>
            <i className="fas fa-envelope" />
          </button>
          <button type="button" onClick={this.onIconClick}>
            <i className="fas fa-user" />
          </button>
        </div>
      </div>
    );
  }
}

export default Header;
