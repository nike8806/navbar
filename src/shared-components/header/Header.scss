
/*TODO MOVE THIS TO GLOBAL VARIABLE*/
$main-color: #FFF;
$secondary-color: #343A40;
$size-factor: 7px;

.flex--nowrap {
  display: flex; 
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-between;
}

// Header styles
.header {
  background: $secondary-color;
  @extend .flex--nowrap;
  padding: $size-factor $size-factor*2;
  color: $main-color;

  &__menu {
    @extend .flex--nowrap;
    padding-top: $size-factor;
    padding-bottom: $size-factor;
    &__title{
      display: none;
    }
    @media only screen and (min-width: 600px) {
      &__title{
        display: block;
      }
    }

    > * {
      padding-left: $size-factor;
      padding-right: $size-factor;
    }
  }

  &__widgets {
    @extend .flex--nowrap;
    align-items: center;

    &__search {
      height: $size-factor*4;
    }

    > * {
      color: $main-color;
      background: none;
      padding-left: $size-factor;
      padding-right: $size-factor;
    }
  }
}
