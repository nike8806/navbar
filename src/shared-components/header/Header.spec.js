import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

let component;

describe('Header component', () => {
  describe('On mount', () => {
    beforeEach(() => {
      component = shallow(
        <Header />
      );
    });

    it('should render correctly', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
