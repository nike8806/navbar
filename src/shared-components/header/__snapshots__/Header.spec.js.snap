// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component On mount should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header__menu"
        >
          <div
            className="header__menu__title"
          >
            Start Bootstrap Menu
          </div>
          <i
            className="fas fa-bars header__menu__bars"
          />
        </div>,
        <div
          className="header__widgets"
        >
          <Header
            classNameComponent="header__widgets__search"
          />
          <button
            onClick={[Function]}
            type="button"
          >
            <i
              className="fa fa-bell"
            />
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            <i
              className="fas fa-envelope"
            />
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            <i
              className="fas fa-user"
            />
          </button>
        </div>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="header__menu__title"
            >
              Start Bootstrap Menu
            </div>,
            <i
              className="fas fa-bars header__menu__bars"
            />,
          ],
          "className": "header__menu",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Start Bootstrap Menu",
              "className": "header__menu__title",
            },
            "ref": null,
            "rendered": "Start Bootstrap Menu",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-bars header__menu__bars",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Header
              classNameComponent="header__widgets__search"
            />,
            <button
              onClick={[Function]}
              type="button"
            >
              <i
                className="fa fa-bell"
              />
            </button>,
            <button
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-envelope"
              />
            </button>,
            <button
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-user"
              />
            </button>,
          ],
          "className": "header__widgets",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "classNameComponent": "header__widgets__search",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fa fa-bell"
              />,
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fa fa-bell",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fas fa-envelope"
              />,
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-envelope",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fas fa-user"
              />,
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-user",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header__menu"
          >
            <div
              className="header__menu__title"
            >
              Start Bootstrap Menu
            </div>
            <i
              className="fas fa-bars header__menu__bars"
            />
          </div>,
          <div
            className="header__widgets"
          >
            <Header
              classNameComponent="header__widgets__search"
            />
            <button
              onClick={[Function]}
              type="button"
            >
              <i
                className="fa fa-bell"
              />
            </button>
            <button
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-envelope"
              />
            </button>
            <button
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-user"
              />
            </button>
          </div>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="header__menu__title"
              >
                Start Bootstrap Menu
              </div>,
              <i
                className="fas fa-bars header__menu__bars"
              />,
            ],
            "className": "header__menu",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Start Bootstrap Menu",
                "className": "header__menu__title",
              },
              "ref": null,
              "rendered": "Start Bootstrap Menu",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-bars header__menu__bars",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Header
                classNameComponent="header__widgets__search"
              />,
              <button
                onClick={[Function]}
                type="button"
              >
                <i
                  className="fa fa-bell"
                />
              </button>,
              <button
                onClick={[Function]}
                type="button"
              >
                <i
                  className="fas fa-envelope"
                />
              </button>,
              <button
                onClick={[Function]}
                type="button"
              >
                <i
                  className="fas fa-user"
                />
              </button>,
            ],
            "className": "header__widgets",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "classNameComponent": "header__widgets__search",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fa fa-bell"
                />,
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fa fa-bell",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fas fa-envelope"
                />,
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-envelope",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fas fa-user"
                />,
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-user",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
